sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala

```diff
@@ -150,10 +154,13 @@ class SparkMetadataOperationSuite extends HiveThriftJdbcTest {
       Seq(
         "CREATE TABLE table1(key INT, val STRING)",
         "CREATE TABLE table2(key INT, val STRING)",
-        "CREATE VIEW view1 AS SELECT * FROM table2").foreach(statement.execute)
+        "CREATE VIEW view1 AS SELECT * FROM table2",
+        "CREATE OR REPLACE TEMPORARY VIEW view_temp_1 AS SELECT 1 as col1",
```

Comment:
@juliuszsompolski We can not show `TEMPORARY VIEW` because it's in different session for some reason.
I have 2 questions here:
1. Do we need to support show `TEMPORARY VIEW` because it does not belong to any database?
2. How do I show `TEMPORARY VIEW` if we support it?

Reply:
I think we should
- show global_temp views, if the database pattern matches the name of the global temp database.
- showing local temp views always (when they match table pattern), but make sure to show them only once.

Local temp views currently do not show up, because in testGetTablesOperation you create a new connection, which is a new session. You could call `statement.getConnection().getMetadata().getTables` directly to stay in the same session.

Reply:
How about temp view also showed in `global_temp`?
![image](https://user-images.githubusercontent.com/5399861/60225476-a3716480-98b9-11e9-8256-e00ffa13616e.png)
![image](https://user-images.githubusercontent.com/5399861/60225490-b1bf8080-98b9-11e9-9946-426ffa52f8a5.png)

Reply:
@wangyum What happens in SQuirreL if you return them without a schema?  I suppose you're asking because it doesn't work?

Reply:
Yes. `TEMPORARY VIEW` won't show if we don't assign an existing schema to it.

Reply:
Hm. The issue with that is that then if someone accesses `global_temp.temp_view_1` in a query, it will result in table not found, as Spark will not find it...
Reporting it with no schema seems fine in beeline/sqlline:
````
0: > create temporary view foo as select * from xxx;
...
0: > !tables
+------------+--------------+------------------------------------+-------------+----------+--+
| TABLE_CAT  | TABLE_SCHEM  |             TABLE_NAME             | TABLE_TYPE  | REMARKS  |
+------------+--------------+------------------------------------+-------------+----------+--+
...
|            |              | foo                                | VIEW        | UNKNOWN  |
+------------+--------------+------------------------------------+-------------+----------+--+
```
so I would stay at that and consider it a SQuirreL problem and postpone handling it better.

Reply:
Yes. I agree with you.